common4j/src/main/com/microsoft/identity/common/components/InMemoryStorageSupplier.kt

```diff
@@ -0,0 +1,63 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.components
+
+import com.microsoft.identity.common.java.cache.IMultiTypeNameValueStorage
+import com.microsoft.identity.common.java.cache.MapBackedPreferencesManager
+import com.microsoft.identity.common.java.interfaces.INameValueStorage
+import com.microsoft.identity.common.java.interfaces.IStorageSupplier
+import com.microsoft.identity.common.java.util.ported.InMemoryStorage
+import java.util.concurrent.ConcurrentHashMap
+
+class InMemoryStorageSupplier : IStorageSupplier {
+
+    private val mNameValueStores: MutableMap<String, INameValueStorage<Any>> =
+        ConcurrentHashMap()
+
+
+    private val mFileStores: MutableMap<String, IMultiTypeNameValueStorage> =
+        ConcurrentHashMap()
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> getNameValueStore(storeName: String, clazz: Class<T>): INameValueStorage<T> {
+        val ret = mNameValueStores[storeName]
+        if (ret != null) {
+            return ret as INameValueStorage<T>
+        }
+
+        val nameValueStore = InMemoryStorage<Any>()
```

Comment:
why "Any"??

Reply:
to match the type in mNameValueStores. (getNameValueStore can be used to get any storage types).